### ISSUE TITLE
Allow different Accounts for each RawContact in insert APIs

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#270]
This should be worked on only after completing #268.

## Problem

When inserting more than one `NewRawContact` in a single insert operation, only one `Account` can be specified for all of the raw contacts in the insert queue. 

## Solution

- [x] Remove `forAccount` function
- [x] Move `account: Account?` declaration to `RawContactEntity` so that it can be added to `NewRawContact`
- [x] Use this account reference for each raw contact in insert APIs

Apply the required changes to the following insert APIs;

- [x] `Insert`
- [x] `ProfileInsert`

Additionally;

- [x] Update sample app
- [x] Update cheatsheet
- [x] Update relevant documentation in code
- [x] Update relevant documentation in docs
- [x] Verify that there are no regressions for related API functions
- [ ] Provide a migration guide as a comment in this issue so that it can be copy-pasted into the release notes

**Comment by vestrel00** [comment]
## Migration guide (to be included in release notes)

[Previously](https://github.com/vestrel00/contacts-android/blob/0.2.4/docs/basics/insert-contacts.md#associating-an-account), to set the Account of `NewRawContact`(s) to be inserted, you would use the `forAccount` function of the `Insert` and `ProfileInsert` APIs,

```kotlin
insert
    .forAccount(account)
    .rawContacts(rawContact)
    .commit()
```

Now, the `forAccount` functions have been removed but you are able to specify the account of each `NewRawContact`,

```kotlin
rawContact.account = account
insert
    .rawContacts(rawContact)
    .commit()
```

Read the new documentation for the full guide; https://vestrel00.github.io/contacts-android/basics/insert-contacts/#associating-an-account

**Comment by vestrel00** [comment]
This is included in [0.3.0](https://github.com/vestrel00/contacts-android/releases/tag/0.3.0)!
